---
title: "class 11"
author: "Vidisha Marwaha (PID: A16677246)"
format: pdf
---

# Section 1. proportion of G/G in a population

Downloaded a CSV file from Ensemble < https://useast.ensembl.org/Homo_sapiens/Variation/Sample?db=core;r=17:39835099-39955100;v=rs8067378;vdb=variation;vf=959672880#373531_tablePanel >

Here we read the CSV file - 

```{r}
mxl <- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(mxl)
```

```{r}
table(mxl$Genotype..forward.strand.)
```

```{r}
(table(mxl$Genotype..forward.strand.) / nrow(mxl)) * 100
```

Now lets look at a different population. I picked the GBR.

```{r}
gbr <- read.csv("373522-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
```

Find proportion of G/G

```{r}
round(table(gbr$Genotype..forward.strand.) / nrow(gbr) *100 ,2)
```

This variant that is associated with childhood asthma is more frequent in the GBR population than MXL population. 

Lets dig into this further.

## Section 4: Population Scale Analysis
[HOMEWORK]
One sample is obviously not enough to know what is happening in a population. You are
interested in assessing genetic differences on a population scale. So, you processed about
~230 samples and did the normalization on a genome level. Now, you want to find whether
there is any association of the 4 asthma-associated SNPs (rs8067378...) on ORMDL3
expression.
This is the final file you got. The first column is sample name, the second
  column is genotype and the third column are the expression values.
 Open a new RMarkdown document in RStudio to answer the following two questions. Submit
 your resulting PDF report with your working code, output and narrative text answering Q13 and
 Q14 to GradeScope.

How many samples do we have?

```{r}
expr <- read.table("rs8067378_ENSG00000172057.6.txt")
head(expr)
```

```{r}
nrow(expr)
```

The sample sizes for each genotype is - 

```{r}
table(expr$geno)
```

```{r}
summary(expr)
```
The median expression levels for each of these genotypes is -
```{r}
genotype <- expr$geno
expression <- expr[, 3] 
genotype_data <- split(expression, genotype)
median_expression <- sapply(genotype_data, median)
median_expression
```


```{r}
library(ggplot2)
```

Let's make a boxplot

```{r}
ggplot(expr) + aes(geno, exp, fill=geno) +
 geom_boxplot(notch=T)
```

The G|G has lower expression compared to A|A according to the boxplot. Having a G|G in this location is associated to having reduced expression of ORMDL3. The SNP does effect the expression of ORMDL3.


